src/shm/sync_util.c

```diff
@@ -0,0 +1,85 @@
+#include "sync_util.h"
+
+#include <string.h>
+#include <sys/mman.h>
+#include <sys/file.h>
+#include <errno.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include "../shared/types.h"
+#include "../shared/file_util.h"
+
+char * lockpath_from_pass(char* passw, char* base){
+    char* path = malloc(strlen(passw) + strlen(base)+ strlen(".lock"));
+    strcpy(path, base);
+    strcat(path, passw);
+    strcat(path, ".lock");
+    return path;
+}
+
+int lock_sync_file(char *passw, char* base){
+    struct stat st = {0};
+    if (stat(base, &st) == -1) {
+        mkdir(base, 0770);
+    }
+
+    char* path = lockpath_from_pass(passw, base);
+
+    int fd = open(path, O_CREAT, 0660);
+    int res = flock(fd, LOCK_EX);
+    if (res){
+        printf("%d %d\n", res, errno);
+        err_and_leave("Error when locking file", 5);
+    }
+    return fd;
+}
+
+void unlink_lock(char *passw, char* base){
+    char* path = lockpath_from_pass(passw, base);
```

Comment:
Danger, danger

Reply:
assuming you mean because I'm not freeing path, this is less of an issue as this cleanup function should only be called once the program ends, but fixed

Reply:
Quite. Then it will be freed anyway as part of the whole memory.